locust/test/test_fasthttp.py

```diff
@@ -0,0 +1,370 @@
+        
+        with self.locust.client.get("/ultra_fast", catch_response=True) as response: pass
+        self.assertEqual(1, self.num_failures)
+        self.assertEqual(1, self.num_success)
+        
+        with self.locust.client.get("/ultra_fast", catch_response=True) as response:
+            raise ResponseError("Not working")
+        
+        self.assertEqual(2, self.num_failures)
+        self.assertEqual(1, self.num_success)
+    
+    def test_catch_response_http_fail(self):
+        with self.locust.client.get("/fail", catch_response=True) as response: pass
+        self.assertEqual(1, self.num_failures)
+        self.assertEqual(0, self.num_success)
+    
+    def test_catch_response_http_manual_fail(self):
+        with self.locust.client.get("/ultra_fast", catch_response=True) as response:
+            response.failure("Haha!")
+        self.assertEqual(1, self.num_failures)
+        self.assertEqual(0, self.num_success)
+        self.assertTrue(
+            isinstance(self.last_failure_exception, CatchResponseError),
+            "Failure event handler should have been passed a CatchResponseError instance"
+        )
+    
+    def test_catch_response_http_manual_success(self):
+        with self.locust.client.get("/fail", catch_response=True) as response:
+            response.success()
+        self.assertEqual(0, self.num_failures)
+        self.assertEqual(1, self.num_success)
+    
+    def test_catch_response_allow_404(self):
+        with self.locust.client.get("/does/not/exist", catch_response=True) as response:
+            self.assertEqual(404, response.status_code)
+            if response.status_code == 404:
+                response.success()
+        self.assertEqual(0, self.num_failures)
+        self.assertEqual(1, self.num_success)
+    
+    def test_interrupt_taskset_with_catch_response(self):
+        class MyTaskSet(TaskSet):
+            @task
+            def interrupted_task(self):
+                with self.client.get("/ultra_fast", catch_response=True) as r:
+                    raise InterruptTaskSet()
+        class MyLocust(FastHttpLocust):
+            host = "http://127.0.0.1:%i" % self.port
+            task_set = MyTaskSet
+        
+        l = MyLocust()
+        ts = MyTaskSet(l)
+        self.assertRaises(InterruptTaskSet, lambda: ts.interrupted_task())
+        self.assertEqual(0, self.num_failures)
+        self.assertEqual(0, self.num_success)
+    
+    def test_catch_response_connection_error_success(self):
+        class MyLocust(FastHttpLocust):
+            host = "http://127.0.0.1:1"
+        l = MyLocust()
+        with l.client.get("/", catch_response=True) as r:
+            self.assertEqual(r.status_code, 0)
+            self.assertEqual(None, r.content)
+            r.success()
+        self.assertEqual(1, self.num_success)
+        self.assertEqual(0, self.num_failures)
+    
+    def test_catch_response_connection_error_fail(self):
```

Comment:
this test for `test_catch_response_connection_error_fail` is identical to `test_catch_response_connection_error_success`.  I think you forgot to make it fail.

Reply:
Oh, good catch! I just copied the tests for the current HTTP client. I will update both tests!